weekly-openmeteo/test_extract.py

```diff
@@ -0,0 +1,123 @@
+'''Contains tests for extract.py'''
+#pylint: disable=W0212
+import json
+from unittest.mock import patch, MagicMock
+import pytest
+from requests import Response
+from extract import request_solar_data,get_county_coordinates,\
+                    convert_to_params
+from api_error import APIError
+
+class TestExtractFunction():
+    '''Contains tests for the extract function'''
+    @patch('extract.get')
+    def test_extract_success(self, mock_get, valid_response,valid_params):
+        '''Tests that a 200 status codes results in the return of the data'''
+        mock_response = Response()
+        mock_response.status_code = 200
+        mock_response._content = json.dumps(valid_response).encode('utf-8')
+        mock_get.return_value = mock_response
+
+        data = request_solar_data(valid_params)
+
+        assert data == valid_response
+
+    @pytest.mark.parametrize("status_code", [400,401,402,403,404,405,408,500])
+    @patch('extract.get')
+    def test_extract_unsuccessful_request(self, mock_get, status_code, valid_params):
```

Comment:
I like the use of parametrisation, allowing the testing of a wide variety of error codes.